Tests/StoreKitUnitTests/ProductsManagerTests.swift

```diff
@@ -161,6 +161,24 @@ class ProductsManagerDiagnosticsTrackingTests: ProductsManagerTests {
         expect(params?.errorCode).to(beNil())
     }
 
+}
+
+// swiftlint:disable type_name
+@available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, *)
+class SK2ProductsManagerDiagnosticsTrackingTests: ProductsManagerTests {
+
+    private var mockDiagnosticsTracker: MockDiagnosticsTracker!
+
+    private var productsManager: ProductsManager!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        try AvailabilityChecks.iOS16APIAvailableOrSkipTest()
+
+        self.mockDiagnosticsTracker = MockDiagnosticsTracker()
+    }
+
     func testFetchProductsWithIdentifiersSK2TracksCorrectly() throws {
```

Comment:
`isStoreKit2Available` is always false for iOS < 16, so this test was running with `storeKit1` even after configuring the manager using SK2